File: data_structures/python/hashmap.py
if __name__ == "__main__":
    # In python, hashmaps are dictionaries
    people = {"John": 32, "Adele": 45, "Bo": 29}

    print(f"John is {people["John"]} years old")
    print(f"Adele is {people["Adele"]} years old")

    people["Jenny"] = 22
    people["Liam"] = 24
    people["Kasper"] = 20
    people["Anja"] = 30

    del people["John"]

    print("\n")

    for person, age in people.items():
        print(f"{person} is {age} years old.")

File: c++/hashmap.cpp
#include <iostream>
#include <map>

int main() {
    std::map<std::string, int> people = { {"John", 32}, {"Adele", 45}, {"Bo", 29} };

    std::cout << "John is: " << people["John"] << "\n";
    std::cout << "Adele is: " << people["Adele"] << "\n";

    people["John"] = 50;

    people.insert({"Jenny", 22});
    people.insert({"Liam", 24});
    people["Kasper"] = 20;
    people["Anja"] = 30;

    people.erase("John");

    for (auto person : people) {
        std::cout << person.first << " is: " << person.second << "\n";
    }

    return 0;
}